### ISSUE TITLE
No method available to add Fields to existing globalLogger variable 

### ISSUE THREAD

**Issue by jajupreetam** [sisyphsu/slf4go#9]
HI,
I want to add few values to Fields of the exiting globalLogger, not to the new one created by GetLogger() or  func NewLogger(name string) . because I see that no possiblity to again set back newly created *Logger  to globalLogger variable.
This is also becasue , slf_core.go always uses globalLogger & not the one I created instance of Logger

func Debug(v ...interface{}) {
...
...
globalLogger.print(DebugLevel, pc[0], nil, v...)
}

Either provide a method to set the newly created Logger to globalLogger or provide new getGlobalLogger()  to get handle to globalLogger so that Fields can be added.

somethig like below in slf_core.go :

func GetGlobalLogger() *Logger {
	return globalLogger
}

OR

func AddFields(fields Fields) {
globalLogger.BindFields(fields Fields)
}

**Comment by sisyphsu** [comment]
The `globalLogger` is used as default logger, I don't think it is a good idea to modify the default logger that other packages may use.

You should create a customized `Logger` in your own package:
```golang
var log = slog.NewLogger()

// ...

func init() {
    log.BindFields(...) // some global fields
}

func yourFunc() {
    log.Info("......") // log with global fields

    log.WithFields(...).Warn("...") // log with global and transient fields to log
}
```

**Comment by jajupreetam** [comment]
My requirement is to log sessionid which is one of the element in the JWT token claims body which comes with every request. So the fields which i have to add to logger is not available by the time when init() gets executed.

Or can you please let me know how to print sid (sessionid) with every log statement without calling log.WithFields(sid value) everytime.

**Comment by sisyphsu** [comment]
> Or can you please let me know how to print sid (sessionid) with every log statement without calling log.WithFields(sid value) everytime.

OK, this is a good question. 

What you need is `Fields` for `goroutine`, like `MDC` in `slf4j`, to bind some variables on `gorouting/thread`, am I correct?

**Comment by sisyphsu** [comment]
This is a very useful feature, but `slf4go` don't support it by now.

I think i could implement it soon.

**Comment by jajupreetam** [comment]
> > Or can you please let me know how to print sid (sessionid) with every log statement without calling log.WithFields(sid value) everytime.
> 
> OK, this is a good question.
> 
> What you need is `Fields` for `goroutine`, like `MDC` in `slf4j`, to bind some variables on `gorouting/thread`, am I correct?

Correct, this is what I need here like MDC in slf4j  to add session Ids/any other fields.

**Comment by sisyphsu** [comment]
This work has done, in `v1.1.0` there has some new functions:
+ `SetContextField(key string, value interface{})`
+ `SetContextFields(fields Fields)`
+ ......

There hasn't docs yet, but I believe you will know how to use them.

You can use those functions to add context variables into goroutine's local storage, and those `Fields` will appear in all log.
